swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/autoscalinggroups/Rules.kt

```diff
@@ -33,13 +34,14 @@ import java.util.Optional
 @Component
 class ZeroInstanceDisabledServerGroupRule(
   private val clock: Clock
-) : Rule<AmazonAutoScalingGroup> {
+) : Rule {
 
   @Value("\${swabbie.resource.disabled.days.count:30}")
   private val disabledDurationInDays: Long = 30
 
-  override fun apply(resource: AmazonAutoScalingGroup): Result {
-    if (resource.isInLoadBalancer()) {
+  override fun <T : Resource> applicableForType(clazz: Class<T>): Boolean = AmazonAutoScalingGroup::class.java.isAssignableFrom(clazz)
```

Comment:
any reason not to use `isInstanceOf` here ?

Reply:
You can use either or. I preferred `isAssignableFrom` because you don't need to know the type of the resource at compile time.